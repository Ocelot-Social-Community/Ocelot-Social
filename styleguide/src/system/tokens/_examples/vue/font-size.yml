#
# FONT-SIZE TOKENS
# Use these tokens to set a font-size.
#

props:
  - name: font-size-xxxx-large
    value: "3rem"
  - name: font-size-xxx-large
    value: "2.5rem"
  - name: font-size-xx-large
    value: "2rem"
  - name: font-size-x-large
    value: "1.5rem"
  - name: font-size-large
    value: "1.25rem"
  - name: font-size-base
    value: "1rem"
  - name: font-size-body
    value: "15px"
  - name: font-size-small
    value: "0.8rem"
  - name: font-size-x-small
    value: "0.7rem"
  - name: font-size-xx-small
    value: "0.6rem"
global:
  type: number
  category: font-size
