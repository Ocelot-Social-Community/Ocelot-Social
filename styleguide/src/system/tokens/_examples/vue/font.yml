#
# FONT TOKENS
# Use these tokens to set font-weight and font-family.
#

props:
  - name: font-family-heading
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-text
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-code
    value: "inconsolata, monospace"
    category: "font-family"

  - name: font-weight-regular
    value: "normal"
    category: "font-weight"
  - name: font-weight-bold
    value: "600"
    category: "font-weight"
global:
  type: ...
  category: font
