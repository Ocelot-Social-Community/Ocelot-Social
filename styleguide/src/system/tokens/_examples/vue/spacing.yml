#
# SPACING TOKENS
# Use these tokens to set padding, margin and position coordinates.
#

aliases:
  xxx-large:
    value: &xxx-large 120
  xx-large:
    value: &xx-large 60
  x-large:
    value: &x-large 40
  large:
    value: &large 28
  base:
    value: &base 20
  small:
    value: &small 12
  x-small:
    value: &x-small 6
  xx-small:
    value: &xx-small 4
  xxx-small:
    value: &xxx-small 2

props:
  - name: xxx-large
    value: *xxx-large
    category: space-size
  - name: xx-large
    value: *xx-large
    category: space-size
  - name: x-large
    value: *x-large
    category: space-size
  - name: large
    value: *large
    category: space-size
  - name: base
    value: *base
    category: space-size
  - name: small
    value: *small
    category: space-size
  - name: x-small
    value: *x-small
    category: space-size
  - name: xx-small
    value: *xx-small
    category: space-size
  - name: xxx-small
    value: *xxx-small
    category: space-size

  - name: space-xxx-large
    value: "{!xxx-large}px"
  - name: space-xx-large
    value: "{!xx-large}px"
  - name: space-x-large
    value: "{!x-large}px"
  - name: space-large
    value: "{!large}px"
  - name: space-base
    value: "{!base}px"
  - name: space-small
    value: "{!small}px"
  - name: space-x-small
    value: "{!x-small}px"
  - name: space-xx-small
    value: "{!xx-small}px"
  - name: space-xxx-small
    value: "{!xxx-small}px"

global:
  type: number
  category: space
