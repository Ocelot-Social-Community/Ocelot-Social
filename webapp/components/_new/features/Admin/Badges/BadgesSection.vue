<template>
  <div class="badge-section">
    <h4>{{ title }}</h4>
    <div class="badge-container">
      <button
        v-for="badge in badges"
        :key="badge.id"
        @click="toggleBadge(badge)"
        :class="{ badge, inactive: !badge.isActive }"
      >
        <img :src="badge.icon" :alt="badge.description" />
      </button>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    title: {
      type: String,
    },
    badges: {
      type: Array,
    },
  },
  methods: {
    toggleBadge(badge) {
      this.$emit('toggleBadge', badge)
    },
  },
}
</script>

<style scoped>
.badge-section h4 {
  margin-bottom: 24px;
}
.badge-container {
  display: flex;
  margin-bottom: 36px;
  flex-flow: row wrap;
  gap: 10px;
}
.badge:hover {
  cursor: pointer;
}
.badge img {
  width: 70px;
}
.badge.inactive {
  opacity: 0.3;
  filter: grayscale(0.4);
}
</style>
