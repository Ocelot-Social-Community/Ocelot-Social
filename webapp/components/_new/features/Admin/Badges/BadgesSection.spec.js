import { render, fireEvent, screen } from '@testing-library/vue'
import BadgesSection from './BadgesSection.vue'

const localVue = global.localVue

const badge1 = {
  id: 'badge1',
  icon: '/icon1',
  type: 'type1',
  description: 'description1',
  isActive: true,
}
const badge2 = {
  id: 'badge2',
  icon: '/icon2',
  type: 'type1',
  description: 'description2',
  isActive: false,
}

describe('Admin/BadgesSection', () => {
  let wrapper

  const Wrapper = (withBadges = true) => {
    return render(BadgesSection, {
      localVue,
      propsData: {
        badges: withBadges ? [badge1, badge2] : [],
      },
      mocks: {
        $t: jest.fn((t) => t),
      },
    })
  }

  describe('without badges', () => {
    beforeEach(() => {
      wrapper = Wrapper(false)
    })

    it('renders', () => {
      expect(wrapper.baseElement).toMatchSnapshot()
    })
  })

  describe('with badges', () => {
    beforeEach(() => {
      wrapper = Wrapper(true)
    })

    it('renders', () => {
      expect(wrapper.baseElement).toMatchSnapshot()
    })

    it('emits toggleButton', async () => {
      const button = screen.getByAltText(badge1.description)
      await fireEvent.click(button)
      expect(wrapper.emitted().toggleBadge[0][0]).toEqual(badge1)
    })
  })
})
