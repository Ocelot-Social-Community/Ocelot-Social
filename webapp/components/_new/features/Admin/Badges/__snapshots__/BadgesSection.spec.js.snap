// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin/BadgesSection with badges renders 1`] = `
<body>
  <div>
    <div
      class="badge-section"
    >
      <h4>
        
      </h4>
       
      <div
        class="badge-container"
      >
        <button
          class="badge"
        >
          <img
            alt="description1"
            src="/api/icon1"
          />
        </button>
        <button
          class="badge inactive"
        >
          <img
            alt="description2"
            src="/api/icon2"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`Admin/BadgesSection without badges renders 1`] = `
<body>
  <div>
    <div
      class="badge-section"
    >
      <h4>
        
      </h4>
       
      <div>
        
    admin.badges.noBadges
  
      </div>
    </div>
  </div>
</body>
`;
